### PR TITLE
adding exception handlers

### PR DESCRIPTION
**Pull request by domino468**

### CODE REVIEW

File: src/main/java/com/coursesPlatform/exceptions/ApiExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ResponseEntity<String> handleIllegalLengthException(Exception e) {
+        return new ResponseEntity<String>("Illegal length !" ,  HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(MailIsAlreadyInUseException.class)
+    ResponseEntity<String> handleEmptyListException(Exception e) {
+        return new ResponseEntity<String>("Mail is already used!", HttpStatus.BAD_REQUEST);
+    }
+    @ExceptionHandler(MailIsInvalidException.class)
+    ResponseEntity<String> handleInvalidMail(Exception e){
```
Comment (LukaszGryziewicz) [inline src/main/java/com/coursesPlatform/exceptions/ApiExceptionHandler.java:32]: Please change the method name to handleInvalidMailException

**Review by LukaszGryziewicz — CHANGES_REQUESTED** [review]

**Review by LukaszGryziewicz — APPROVED** [review]

File: src/main/java/com/coursesPlatform/coursePortfolio/CustomerService.java [hunk 2]
```diff
@@ -22,7 +22,7 @@ CustomerDTO add(CustomerDTO customerDTO) {
         Optional<Customer> customerByMail = customerRepository
                 .findCustomerByMail(customerDTO.getMail());
         if(!EmailValidator.getInstance().isValid(customerDTO.getMail())){
-            throw new MailIsInvalid();
+            throw new MailIsInvalidException();
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/CustomerService.java:25]: How about `InvalidMailException`?

File: src/main/java/com/coursesPlatform/coursePortfolio/TrainerNotFoundException.java [hunk 3]
```diff
@@ -2,5 +2,6 @@
 
 public class TrainerNotFoundException extends RuntimeException {
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/TrainerNotFoundException.java:3]: Why `RuntimeException` not `Exception`?

File: src/main/java/com/coursesPlatform/exceptions/ApiExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+public class ApiExceptionHandler {
+
+    @ExceptionHandler(value = {CategoryNotFoundException.class, CourseNotFoundException.class, TrainerNotFoundException.class, CustomerNotFoundException.class})
+    ResponseEntity<String> handleNotFoundExceptions(Exception e) {
+        return new ResponseEntity<String>("Not Found!", HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(IllegalLengthException.class)
+    ResponseEntity<String> handleIllegalLengthException(Exception e) {
+        return new ResponseEntity<String>("Illegal length !" ,  HttpStatus.BAD_REQUEST);
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/exceptions/ApiExceptionHandler.java:24]: I think it should be specified which length is illegal. What is accepted length?

File: src/test/java/com/coursesPlatform/coursePortfolio/CustomerTest.java [hunk 5]
```diff
@@ -81,6 +81,6 @@ public void shouldThrowExceptionWhenMailIsInvalid() {
         //when
         Throwable thrown = catchThrowable(()-> customerService.add(customer));
         //then
-        assertThat(thrown).isInstanceOf(MailIsInvalid.class);
+        assertThat(thrown).isInstanceOf(MailIsInvalidException.class);
```
Comment (adamkulawik) [inline src/test/java/com/coursesPlatform/coursePortfolio/CustomerTest.java:84]: Consider using something described at https://www.baeldung.com/assertj-exception-assertion

**Pull request merged**